WordPress/Classes/Stores/StatsInsightsStore.swift

```diff
@@ -981,3 +985,66 @@ private extension InsightStoreState {
     }
 
 }
+
+// MARK: - iOS 14 Widgets Data
+private extension InsightStoreState {
+
+    private func saveHomeWidgetTodayData(stats: TodayWidgetStats) {
+
+        guard let siteID = SiteStatsInformation.sharedInstance.siteID else {
+            return
+        }
+
+        var homeWidgetTodayCache = HomeWidgetTodayData.read() ?? initializeHomeWidgetTodayData()
```

Comment:
In reference to `HomeWidgetTodayData.read()`, `HomeWidgetTodayData.write()` and `HomeWidgetTodayData.delete()`

I noticed that, at each call of these three methods, we're instantiating `HomeWidgetCache<HomeWidgetTodayData>`.

Any thoughts on making `HomeWidgetTodayData` instantiable (instead of purely static) and initializable with a specific  `HomeWidgetCache<HomeWidgetTodayData>` (since it's not going to change anyway)?

For more context on why I'm suggesting this: I'm not a big fan of static methods like these ones because then you kinda have a dependency with a specific class that's harder to change or break in the future.  It also tends to make it slightly harder to test the code, since it can't be replaced through a dependency injection easily.  Just my 2 cents.

Reply:
I don't like static methods either, and I'm open to refactor this, maybe in a future PR, as we are still working on the overall new widget architecture.
A few considerations, for more context:
- `HomeWidgetTodayData` is indeed instantiable, because it's the `TimelineEntry` used by the widgets. Those static methods could move elsewhere though
- `HomeWidgetCache` is basically a wrapper for `Filemanager.default` (a singleton), that only does the job of telling `Filemanager` what is the file to handle and how to encode the specific `HomeWidgetTodayData`. Since the file is shared between the widget and the app, we could try to inject an instance of an object on both `StatsInsightStore` and the widget, I'd just do it in a future PR if you'd agree.
- Still not ideal, but to offer a bit more testability, I made the cache object injectable in the static method.